java/src/jmri/jmrit/display/layoutEditor/LayoutEditor.java

```diff
@@ -2021,14 +2027,39 @@ protected void targetWindowClosingEvent(java.awt.event.WindowEvent e) {
      * @param inValidateMode boolean: if true, valid text == green, invalid text
      *                       == red background; if false, valid text == green,
      *                       invalid text == yellow background
-     * @param inEnable       boolean to enable / disable the JmriBeanComboBoxes
+     * @param inEnable       boolean to enable / disable the JmriBeanComboBox
      */
     public void setupComboBox(JmriBeanComboBox inComboBox, boolean inValidateMode, boolean inEnable) {
         inComboBox.setEnabled(inEnable);
         inComboBox.setEditable(true);
         inComboBox.setValidateMode(inValidateMode);
         inComboBox.setText("");
+        int c = Math.max(8, inComboBox.getItemCount());
```

Comment:
Why have the while loop when you can use Math.min() to ensure the size of your combobox is capped correctly?